runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala

```diff
@@ -222,18 +224,33 @@ final class GrpcClientSettings private (
   def withDefaultPort(value: Int): GrpcClientSettings = copy(defaultPort = value)
   def withCallCredentials(value: CallCredentials): GrpcClientSettings = copy(callCredentials = Option(value))
   def withOverrideAuthority(value: String): GrpcClientSettings = copy(overrideAuthority = Option(value))
+
+  /** Note: Netty client backend specific setting */
   def withSslProvider(sslProvider: SslProvider): GrpcClientSettings =
     Option(sslProvider).fold(this)(sslProvider => copy(useTls = true, sslProvider = Some(sslProvider)))
-
-  /**
-   * Prefer using `withContextManager`: withSslContext forces the ssl-provider 'jdk', which is known
-   * not to work on JDK 1.8.0_252.
-   */
   def withSslContext(sslContext: SSLContext): GrpcClientSettings =
     Option(sslContext).fold(this)(sslContext => copy(useTls = true, sslContext = Option(sslContext)))
   def withTrustManager(trustManager: TrustManager): GrpcClientSettings =
     Option(trustManager).fold(this)(trustManager => copy(useTls = true, trustManager = Option(trustManager)))
 
+  /**
+   * Scala API: For each new client connection, invoke this provider to set up TLS, useful for example for using rotating
+   * certs for the client provided by `SSLContextFactory.refreshingSSLEngineProvider`.
+   *
+   * When setting this the other TLS settings (`sslContext`,`tlsManager`) must not be set.
+   */
+  def withSslContextProvider(provider: () => SSLContext): GrpcClientSettings =
+    copy(useTls = true, sslContextProvider = Some(provider))
+
+  /**
+   * Java API: For each new client connection, invoke this provider to set up TLS, useful for example for using rotating
+   * certs for the client provided by `SSLContextFactory.refreshingSSLEngineProvider`.
+   *
+   * When setting this the other TLS settings (`sslContext`,`tlsManager`) must not be set.
+   */
+  def withSslContextProvider(provider: akka.japi.function.Creator[SSLContext]): GrpcClientSettings =
```

Comment:
no risk that those two Scala and Java signatures are somehow the same?

Reply:
Good catch, they probably end up problematic with SAMs, not sure why examples didn't trigger a warning

Reply:
Verified, what doesn't work is a call in Java with a lambda, then it becomes ambiguous, the sample code explicitly used a `Creator<SSLContext>` created in a method so I didn't notice.